### PR TITLE
add At_mul_B!

### PR DESCRIPTION
**Pull request by tanmaykm**
adding `At_mul_B!`, riding on `A_mul_B!`

### CODE REVIEW

**Comment by ViralBShah** [comment]
This seems to just do the transpose of A and multiply it by B? Doesn't the fallback work?

**Comment by andreasnoack** [comment]
Can't you mimic `Ac_mul_B!` to avoid the explicit transpose. @ViralBShah We don't have fallbacks for the inplace versions.

**Comment by tanmaykm** [comment]
Yes, it will be better to do as in `Ac_mul_B!`. Will update.

**Comment by tanmaykm** [comment]
@andreasnoack updated with a common method. Is this ok?

File: src/DistributedArrays.jl [hunk 1]
```diff
@@ -1143,24 +1143,42 @@ end
...
-                    localpart(A)*convert(localtype(B), Bjk)
+            Acuts = A.cuts[Ad2]
+            Ccuts = C.cuts[2]
+            Bjk = B[Acuts[j]:Acuts[j + 1] - 1, Ccuts[k]:Ccuts[k + 1] - 1]
+            for i = 1:size(A.pids, Ad1)
+                p = (tA == 'N') ? procs(A)[i,j] : procs(A)[j,i]
+                R[i,j,k] = remotecall(p) do
+                    Atc = localpart(A)
+                    if tA == 'T'
+                        Atc = transpose(Atc)
```
Comment (andreasnoack) [inline src/DistributedArrays.jl]: Please don't make explicit transposes. Use `return Atc.'*convert(localtype(B), Bjk)` instead.

**Comment by andreasnoack** [comment]
Makes sense to combine the methods. Would you also add some tests?

File: src/DistributedArrays.jl (hunk 1 above)
Comment (tanmaykm) [inline src/DistributedArrays.jl]: updated. why is `.'` preferred over `transpose`?

Comment (andreasnoack) [inline src/DistributedArrays.jl]: The reason is that
```jl
julia> expand(:(A'*B))
:(Ac_mul_B(A,B))      
```
and `Ac_mul_B` eventually calls [`xgemm`](http://www.netlib.org/lapack/explore-html/d1/d54/group__double__blas__level3.html#gaeda3cbd99c8fb834a60a6412878226e1) which computes `C:=α*A'B + β*C` by updating `C` in place such that `A` is never transposed.

Comment (tanmaykm) [inline src/DistributedArrays.jl]: Ah ok. Thanks.

**Comment by tanmaykm** [comment]
Added some tests.

**Comment by andreasnoack** [comment]
Great.

**Pull request merged**